test/jdk/java/awt/image/MemoryLeakTest/MemoryLeakTest.java

```diff
@@ -22,154 +22,100 @@
  */
 
 /* @test
-   @bug 4078566 6658398
-   @summary Test for a memory leak in Image.
-   @run main/manual MemoryLeakTest
-*/
-
-import java.applet.Applet;
-import java.lang.*;
-import java.awt.*;
-import java.awt.event.*;
-
-class Globals {
-  static boolean testPassed=false;
-  static Thread mainThread=null;
-}
-
-public class MemoryLeakTest extends Applet {
-
-public static void main(String args[]) throws Exception {
-  new TestDialog(new Frame(), "MemoryLeakTest").start();
-  new MemoryLeak().start();
-  Globals.mainThread = Thread.currentThread();
-  try {
-    Thread.sleep(300000);
-  } catch (InterruptedException e) {
-    if (!Globals.testPassed)
-      throw new Exception("MemoryLeakTest failed.");
-  }
-}
-
-}
-
-class TestDialog extends Dialog
-    implements ActionListener {
-
-TextArea output;
-Button passButton;
-Button failButton;
-String name;
-
-public TestDialog(Frame frame, String name)
-{
-  super(frame, name + " Pass/Fail Dialog");
-  this.name = name;
-  output = new TextArea(11, 50);
-  add("North", output);
-  output.append("Do the following steps on Solaris only.\n");
-  output.append("Maximize and minimize the Memory Leak Test window.\n");
-  output.append("Execute the following after minimize.\n");
-  output.append("    ps -al | egrep -i 'java|PPID'\n");
-  output.append("Examine the size of the process under SZ.\n");
-  output.append("Maximize and minimize the Memory Leak Test window again.\n");
-  output.append("Execute the following after minimize.\n");
-  output.append("    ps -al | egrep -i 'java|PPID'\n");
-  output.append("Examine the size of the process under SZ.\n");
-  output.append("If the two SZ values are the same, plus or minus one,\n");
-  output.append("then click Pass, else click Fail.");
-  Panel buttonPanel = new Panel();
-  passButton = new Button("Pass");
-  failButton = new Button("Fail");
-  passButton.addActionListener(this);
-  failButton.addActionListener(this);
-  buttonPanel.add(passButton);
-  buttonPanel.add(failButton);
-  add("South", buttonPanel);
-  pack();
-}
-
-public void start()
-{
-  show();
-}
+ * @bug 4078566 6658398
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @summary Test for a memory leak in Image.
+ * @run main MemoryLeakTest
+ */
 
-public void actionPerformed(ActionEvent event)
-{
-    if ( event.getSource() == passButton ) {
-      Globals.testPassed = true;
-      System.err.println(name + " Passed.");
-    }
-    else if ( event.getSource() == failButton ) {
-      Globals.testPassed = false;
-      System.err.println(name + " Failed.");
+import java.awt.Color;
+import java.awt.Frame;
+import java.awt.Graphics;
+import java.awt.Image;
+import java.awt.event.ComponentEvent;
+import java.awt.event.ComponentListener;
+
+public class MemoryLeakTest {
+    private static final String INSTRUCTIONS =
+        """
+         Do the following steps on Solaris only.
```

Comment:
Since the test can be run on Unix platforms, how about including it in the instructions. 
```suggestion
         Do the following steps on Unix platforms.
```

Reply:
Fixed